shapeways/oauth2_client.py

```diff
@@ -51,18 +75,57 @@ def authenticate(self, client_id, client_secret):
     def _validate_response(self, response):
         """
         Internal function - validate results
-        :rtype: list()
-        """
-        if response.status_code != 200:
-            raise RuntimeError("Call threw status {}".format(response.status_code))
+        :rtype dict()
+        """
+
+        # Default values - these will be modified as needed by the rate limiting error handling
+        rate_limit = RateLimitingHeaders(
+            rate_limit_type=SW_RATE_LIMIT,
+            is_rate_limited=False
+        )
+        headers = response.headers
+
+        # Handle HTTP errors
+        if response.status_code != requests.codes.ok:
+            if response.status_code == requests.codes.too_many_requests:
+                # We're rate limited
```

Comment:
maybe add "by either Cloudflare or Shapeways" for clarity

Reply:
That comes later down in the code.

Reply:
true